analyzers/src/SonarAnalyzer.CSharp/Rules/StaticFieldVisible.cs

```diff
@@ -34,22 +34,27 @@ public sealed class StaticFieldVisible : SonarDiagnosticAnalyzer
             context.RegisterNodeAction(
                 c =>
                 {
-                    var fieldDeclaration = (FieldDeclarationSyntax)c.Node;
-                    foreach (var diagnostic in GetDiagnostics(fieldDeclaration.Declaration, c.SemanticModel))
+                    foreach (var diagnostic in GetDiagnostics((FieldDeclarationSyntax)c.Node, c.SemanticModel))
                     {
                         c.ReportIssue(diagnostic);
                     }
                 },
                 SyntaxKind.FieldDeclaration);
 
-        private static IEnumerable<Diagnostic> GetDiagnostics(VariableDeclarationSyntax declaration, SemanticModel semanticModel) =>
-            declaration.Variables
-                       .Where(x => FieldIsRelevant(semanticModel.GetDeclaredSymbol(x) as IFieldSymbol))
-                       .Select(x => Diagnostic.Create(Rule, x.Identifier.GetLocation(), x.Identifier.ValueText));
+        private static IEnumerable<Diagnostic> GetDiagnostics(FieldDeclarationSyntax declaration, SemanticModel semanticModel) =>
+            FieldIsRelevant(declaration)
+            ? declaration.Declaration.Variables
+                    .Where(x => !FieldIsThreadSafe(semanticModel.GetDeclaredSymbol(x) as IFieldSymbol))
+                    .Select(x => Diagnostic.Create(Rule, x.Identifier.GetLocation(), x.Identifier.ValueText))
+            : Enumerable.Empty<Diagnostic>();
 
-        private static bool FieldIsRelevant(IFieldSymbol fieldSymbol) =>
-            fieldSymbol is {IsStatic: true, IsConst: false, IsReadOnly: false}
-            && fieldSymbol.DeclaredAccessibility != Accessibility.Private
-            && !fieldSymbol.GetAttributes().Any(x => x.AttributeClass.Is(KnownType.System_ThreadStaticAttribute));
+        private static bool FieldIsRelevant(FieldDeclarationSyntax node) =>
+            node.Modifiers.Any(SyntaxKind.StaticKeyword)
```

Comment:
If you invert the order of `node.Modifiers.Count > 1` and `node.Modifiers.Any(SyntaxKind.StaticKeyword)`, you have an O(1) check in all cases where there's a single modifier, which is better than going through all the modifiers at least once.

That actually makes almost no difference whatsoever, since the number of modifiers is limited and, unlike in C++, where you can do `public: public: public: public: int a;`), C# seems to raise a compiler error when you repeat many times the same modifier.